### PR TITLE
feat(robot-server): Robot server logging and settings

### LINKED ISSUE #5013: API configuration file
## overview

- `robot-server` configuration for the moment is just a handful of items and they should be moved from command line args and into their own json/yaml file.
- This includes port, host, and logging setup for robot-server

### PR DESCRIPTION
**Pull request by amitlissack**
## overview

Logging configuration for robot-server that pipes logs to "opentrons.api" syslog handle.

Settings are read from environment not command line. Used Pydantic's BaseSettings to create a settings object that automatically overrides using environment variables. 

Additionally, using `dotenv` to optionally load environment variables from a file.

The approach is that the default settings are the settings for the robot environment.  A `dev.env` is used to run locally.

## changelog

- Adds logging configuration for robot-server
- Replaces command line arguments with a settings defined in `RobotServerSettings`. 

## risk assessment

Low risk. The only change that affects production code is use of settings to configure how app starts up and chats with nginx.

closes #5013

### CODE REVIEW

File: robot-server/robot_server/service/logging.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # TODO Amit 2019/04/08 Move the logging level from robot configs to
+    #  robot server mutable configs.
+    robot_conf = robot_configs.load()
+    level = logging._nameToLevel.get(robot_conf.log_level, logging.INFO)
+
+    if IS_ROBOT:
+        c = _robot_log_config(level)
+    else:
+        c = _dev_log_config(level)
+    dictConfig(c)
```
Comment (sfoster1) [inline robot-server/robot_server/service/logging.py:17]: this should also call `api.util.logging_config`, which sets up the separate api and serial loggers

Comment (amitlissack) [reply robot-server/robot_server/service/logging.py:17]: This only initializes the robot_server logging. `api.util.logging_config` is already called when we initialize hardware.

Comment (sfoster1) [reply robot-server/robot_server/service/logging.py:17]: nevermind i can't read

**Comment by amitlissack** [comment]
> Works ok once I get dotenv installed. One thing though - this sets up the fastapi access logs to the same logging destination as the rest of the api logs, which could be super spammy. We should probably lower their level to `debug` or send them to a different syslog target.

I had wondered if the `robot-server` and uvicorn crap should go to `opentrons.api` or their own syslog target. I feel like they should be together.

File: robot-server/Makefile [hunk 2]
```diff
@@ -140,3 +139,10 @@ change-left:
 .PHONY: term
 term:
 	ssh -i $(br_ssh_key) $(ssh_opts) root@$(host)
+
+.PHONY: docs
+docs: settings_schema.json
+
+settings_schema.json: robot_server/settings.py
+	$(python) settings_schema.py
```
Comment (mcous) [inline robot-server/Makefile:147]: Can this generation file live in a directory called `scripts` for organizational consistency?

Comment (amitlissack) [reply robot-server/Makefile:147]: Good idea.

File: robot-server/robot_server/settings.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+@lru_cache(maxsize=1)
+def get_settings() -> 'RobotServerSettings':
+    """Get the settings"""
+    load_dotenv(get_dotenv_path())
+    return RobotServerSettings()
+
+
+def get_dotenv_path() -> Path:
+    """Get the location of the settings file"""
+    environment = "robot" if IS_ROBOT else "dev"
```
[the remark this answers is not in the document]
Comment (amitlissack) [reply robot-server/robot_server/settings.py:24]: This is good feedback. I had a hard time with this.

**Pull request merged**